File: codegen_typescript/src/sample.ts
import { execSync } from "child_process";
import { Asap2File, Characteristic } from "./a2l";
import * as fs from 'fs';
import * as iconv from 'iconv';

function parseA2lFile(jar: string, a2lPath: string):Asap2File {
    let output: string = execSync("java -jar \"" + jar + "\" -mj -a2l \"" + a2lPath + "\"").toString();
    let a2lFile: Asap2File = JSON.parse(output);
    return a2lFile;
}

let a2lFile = null;
if(process.argv[2].endsWith(".jar")) {
    a2lFile = parseA2lFile(process.argv[2], process.argv[3]);
} else {
    let ic = new iconv.Iconv('iso-8859-1', 'utf-8');
    let content = ic.convert(fs.readFileSync(process.argv[2])).toString();
    a2lFile = JSON.parse(content);
}
for (let c of a2lFile.project?.modules![0].characteristics!) {
    console.log(c.name);
}

File: codegen_python/sample.py
import a2l
import subprocess
import sys

a2lFile = None
if(sys.argv[1].endswith(".jar")):
    proc = subprocess.Popen([
        "java",
        "-jar",
        sys.argv[1],
        "-mj",
        "-c", "ISO-8859-1",
        "-a2l",
        sys.argv[2],
    ], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    rawJson, err = proc.communicate()
    jsonContent = rawJson.decode("ISO-8859-1")
    a2lFile = a2l.Asap2File.parse_raw(jsonContent)
elif(sys.argv[1].endswith(".json")):
    a2lFile = a2l.Asap2File.parse_file(sys.argv[1], encoding="ISO-8859-1")

for c in a2lFile.project.modules[0].characteristics:
    print(c.name)